### PR TITLE
Event layers filters

### PR DESCRIPTION
**Pull request by satellitestudiodesign**
- [x] remove dataset mock

Screenshot 2023-05-31 at 15 09 35
Screenshot 2023-05-31 at 15 09 41

### CODE REVIEW

File: libs/dataviews-client/src/resolve-dataviews-generators.ts [hunk 1]
```diff
@@ -148,6 +148,10 @@ export function getGeneratorConfig(
         dataview,
         DatasetTypes.Events
       )
+
+      //TODO console.log('dataview:', dataview)
```
Comment (j8seangel) [inline libs/dataviews-client/src/resolve-dataviews-generators.ts:152]: 👀

File: libs/layer-composer/src/generators/tile-cluster/tile-cluster.ts [hunk 2]
```diff
@@ -32,6 +32,15 @@ class TileClusterGenerator {
         Array.isArray(config.eventTypes) ? config.eventTypes.join(',') : config.eventTypes
       )
     }
+
+    if (config.filters?.flag) {
```
Comment (j8seangel) [inline libs/layer-composer/src/generators/tile-cluster/tile-cluster.ts:36]: What about making this more flexible with something like:
```
Object.values(config.filters).forEach(([filter, value]) => url.searchParams.set(filter, value)
```

File: libs/layer-composer/src/generators/types.ts [hunk 3]
```diff
@@ -249,6 +249,10 @@ export interface TileClusterGeneratorConfig extends GeneratorConfig {
    * Event id to highlight on active
    */
   currentEventId?: string
+  /**
+   * Filter the points
+   */
+  filters?: any
```
Comment (j8seangel) [inline libs/layer-composer/src/generators/types.ts:255]: I'd add a basic type definition like `Record<string, any>`

File: libs/ui-components/src/index.ts [hunk 4]
```diff
@@ -20,6 +20,7 @@ export * from './radio'
 export * from './select'
 export * from './split-view'
 export * from './slider'
+export * from './slider-range'
```
Comment (j8seangel) [inline libs/ui-components/src/index.ts:23]: 👏

File: apps/fishing-map/features/workspace/activity/ActivitySchemaFilter.tsx [hunk 5]
```diff
@@ -65,23 +83,25 @@ function ActivitySchemaFilter({
...
     (rangeSelected) => {
       const filterRange = getRangeLimitsBySchema(schemaFilter)
       if (rangeSelected[0] === filterRange[0] && rangeSelected[1] === filterRange[1]) {
         onClean(id)
+      } else if (!Number.isNaN(rangeSelected)) {
+        const value = unit ? VALUE_TRANSFORMATIONS_BY_UNIT[unit].out(rangeSelected) : rangeSelected
+        onSelect(id, value, true)
       } else {
-        const selection = rangeSelected.map((id) => ({
+        const selection = rangeSelected.flatMap((id) => ({
```
Comment (j8seangel) [inline apps/fishing-map/features/workspace/activity/ActivitySchemaFilter.tsx:97]: What is the difference to use flatMap here?

**Review by j8seangel — COMMENTED** [review]
Great feature! once we merge this https://github.com/GlobalFishingWatch/gfw-terraform-api/pull/212 you will need to update [this](https://github.com/GlobalFishingWatch/frontend/pull/2083/files#diff-2b346a7d39e049269fe3b0e01d673c5fefda9ca823f118cf80e4a05ba926cd79R116) and remove the mocked dataset

File: libs/layer-composer/src/generators/tile-cluster/tile-cluster.ts (hunk 2 above)
Comment (satellitestudiodesign) [reply libs/layer-composer/src/generators/tile-cluster/tile-cluster.ts:36]: I tried but the events endpoint expects 'flags' and the filter needs to be called flag to pull the flag list

**Pull request merged**